### PR TITLE
feat:  DCMAW-15957:  Make the corner radius of GDSCardView configurable

### PR DESCRIPTION
**Pull request by joshdubey**
# Make the corner radius of GDSCardView configurable

Added the abillty to specify a corner radius on card views, keeping a default value of DesignSystem.CornerRadius.card.  This was done to support migration of Wallet screens to the Design System

https://govukverify.atlassian.net/browse/DCMAW-15957

# Checklist

## Before raising your pull request:
- [x] Ran and tested the app locally ensuring it builds
- [x] Pull request has a clear title with ticket ID and a short description about the feature or update

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
- [x] Written Unit and Integration tests if needed

- [x] Met all accessibility requirements?
(VoiceOver, DynamicType and using a keyboard for navigation)

## Before merging your pull request:
- [ ] Ran the app to ensure that no regressions have been caused by changes during code review.
- [ ] Targeted the correct branch; `develop`, `release` or `main`

### CODE REVIEW

File: Tests/Components/GDSCard/GDSCardTests.swift [hunk 1]
```diff
@@ -18,7 +18,25 @@ struct GDSCardViewTests {
...
+
+        #expect(cardStackView.spacing == .zero)
+        #expect(cardStackView.alignment == .fill)
+        #expect(cardStackView.distribution == .fill)
+        #expect(cardStackView.layer.cornerRadius == 0)
+        #expect(cardStackView.layer.masksToBounds == true)
+        #expect(cardStackView.translatesAutoresizingMaskIntoConstraints == false)
+        #expect(cardStackView.backgroundColor == DesignSystem.Color.Backgrounds.card)
+    }
+
```
Comment (krissbennett-dd) [inline Tests/Components/GDSCard/GDSCardTests.swift:39]: Add a test to ensure that when we don't pass a corner radius that it defaults to the expected value we have before this change.

**Review by krissbennett-dd — CHANGES_REQUESTED** [review]
Additional test needed

File: Tests/Components/GDSCard/GDSCardTests.swift (hunk 1 above)
Comment (joshdubey) [reply Tests/Components/GDSCard/GDSCardTests.swift:39]: This is already covered by the previous test ("Check configuration of the outer stack view")

**Review by krissbennett-dd — APPROVED** [review]
